### ISSUE TITLE
Integrate with Kubernetes

### ISSUE THREAD

**Issue by mraerino** [testcontainers/testcontainer-go#22]
Someone at a meetup lately suggested it would be cool to be able to start containers on Kubernetes so blackbox tests of containers can be done using standard test libraries.

Since this library is in an early state, it would be a great opportunity to abstract the container layer in such a way that it does not matter whether the Docker or Kubernetes APIs are used.

**Comment by gianarb** [comment]
Yep, this is a good idea, I think we should create a Container interface with the current functions we implemented in the container like:

```
type Container interface{
     Terminate(ctx context.Context, t *testing.T) error
     LivenessCheckPorts(ctx context.Context) (nat.PortSet, error) 
     GetIPAddress(ctx context.Context) (string, error)
     GetHostEndpoint(ctx context.Context, port string) (string, string, error)
}
```
And we can have a new file `kubernetes.go` that implements that interface with a function `RunPod` (it should return a container just like [`RunContainer`](https://github.com/testcontainers/testcontainer-go/blob/master/docker.go#L109) but via kubernetes (a pod).

Nice idea! What do you think?

**Comment by mraerino** [comment]
That sounds great!

**Comment by mraerino** [comment]
The foundation for this has been laid in #36

**Comment by rnorth** [comment]
This is probably something we should do with the Java implementation and other forks, at the right time. Please could we discuss the general approach to tackling this?

I can't think of anything specific, but it would be good for all the implementations to be gradually moving towards convergence on feature sets rather than divergence 😄 .

**Comment by gianarb** [comment]
We have a lot to fix before this. So yes! Let's work on this together when
everyone will be ready.

**Comment by mdelapenya** [comment]
I think it makes sense to move this thread to Discussions
